### ISSUE TITLE
Write mutation tests for all bip62 vectors + JoinSplit ecdsa sig mutations.

### ISSUE THREAD

**Issue by nathan-at-least** [zcash/zcash#1188]
This issue came out of https://github.com/zcash/zcash/pull/1183#issuecomment-238049883

> I'd like to see more exhaustive mutation tests that check each of a list of known malleability vectors, such as all 9 cases here: https://github.com/bitcoin/bips/blob/master/bip-0062.mediawiki
>
> Additionally, we should test malleability of the JoinSplit signatures (eg: sign flips).
>
> Even though it should be obvious that these won't impact this "stable txid" having these tests would make me feel better.

**Comment by daira** [comment]
I think we already have nonmalleability tests for the JoinSplit signatures.

**Comment by str4d** [comment]
We also already contain the fix to BIP 66 case 1 (enforcing strict DER encoding), because that was added in Bitcoin 0.8. However, for all other cases we would be writing failing tests (until either we write fixes, or we merge them from upstream). Thus this ticket blocks on #1433.

**Comment by str4d** [comment]
Unlikely at this stage that this will be done for beta 2. I might work on this on Sunday/Monday once I'm in the US, if there isn't more important work for beta 2.

**Comment by str4d** [comment]
Not done for beta 2.

**Comment by bitcartel** [comment]
#1433 is unblocked.  Writing tests for things which haven't been fixed is low priority for launch, so moving to 1.0.1

**Comment by nathan-at-least** [comment]
Next step: check which bip62 vectors are not yet covered.

Also related: #2360
